source/common/upstream/dynamic_host_filter_state.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "envoy/stream_info/filter_state.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class DynamicHostFilterState : public StreamInfo::FilterState::Object {
```

Comment:
On a second thought, shall we just use StringAccessor instead of a customized filter state here as it just stores a string? That give the setter more flexibility (e.g. via wasm).

Reply:
Can you point to an example of this StringAccessor? I'm not familiar with what you are referring. In any way, I followed the same pattern as other filters are using to add custom behavior, such as the TCP proxy that uses filter state to dynamically select the upstream cluster. I believe we should follow the same pattern

Reply:
StringAccessor is [here](https://github.com/envoyproxy/envoy/blob/main/envoy/router/string_accessor.h). It is still the same pattern but not define a customized object to store the data.

Reply:
You should still use the key name `envoy.upstream.dynamic_host` but the type of filter object should be a StringAccessor and I don't see why it needs to be a custom type. A string accessor is easier to get/set by other filters and for [logging and header substitution](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-filter-state)

Reply:
Done